rfcs/text/0024-ticket-fields.md

```diff
@@ -67,6 +116,25 @@ The goal here is to research and understand the impact of these changes on users
 Stage 1: Identify potential concerns, implementation challenges, or complexity. Spend some time on this. Play devil's advocate. Try to identify the sort of non-obvious challenges that tend to surface later. The goal here is to surface risks early, allow everyone the time to work through them, and ultimately document resolution for posterity's sake.
 -->
 
+- **Ingest:** Extracting the details of tickets from these systems could potentially prove to be a challenge.  All are backed by a DB
+and organizations are usually able to extract details from them to another source wherein they generally do their reporting.
+These alternate sources are typically SQL databases themselves.  Accordingly, it should be possible to attach an ETL process
+to these DB sources to overcome this challenge.
+
+- **Synchronization of tickets:**  Tickets are not static documents, they change as they progress through their lifecycle.  The timing,
+handling, and application of updates from the source to Elasticsearch would have to be cared for.  Too much delay, and the
+benefits of a centralized Enterprise Search solution are lost; too many updates and teh Elasticsearch instance can suffer performance
+challenges and/or become too costly to manage as it has to be scaled to support the load.
+
+- **Index management:** Enterprise Search Use Cases usually rely on querying indexes of static data and not time series data.
+With tickets, they could be considered static data, however, the number of unique documents keeps growing, and some sort
+of roll over policy has to be applied.  They could be considered time series because tickets are created at a point in time,
+and therefore could be aligned to a time series index.
+
+- **Ticket correlation:** As each type of ticket is its own distinct and separate document, it doesn't necessarily indicate a
+relationship to any other ticket.  This is where things like `tags` comes into play to create an association between the
+various types of tickets and what other common associations they may have for things like sprints, releases, etc.
+
```

Comment:
Hi @kc-comcast on the Topic of Static versus Time Series Data

Do you view a Ticket as a History of Transactions, that is to say a series of Time series events connected by e.g. 

`ticket.number`

?

In this case you might want to consider adding

`ticket.updated`

And a rule to ensure that at least one (the most recent) of 

`ticket.created`
`ticket.updated`

is present and available to be copied to `@timestamp` 

This would align the data well with the Timeseries pattern and enable the user to apply both off-the-shelf and custom analytics using Kibana.

Reply:
> Hi @kc-comcast on the Topic of Static versus Time Series Data
> 
> Do you view a Ticket as a History of Transactions, that is to say a series of Time series events connected by e.g.
> 
> `ticket.number`
> 
> ?
> 
> In this case you might want to consider adding
> 
> `ticket.updated` (or `ticket.last_update`)
> 
> And a rule to ensure that at least one (the most recent) of
> 
> `ticket.created`
> `ticket.updated`
> 
> is present and available to be copied to `@timestamp`
> 
> This would align the data well with the Timeseries pattern and enable the user to apply both off-the-shelf and custom analytics using Kibana.

This was the challenge.  In terms of best Use Case for ECS, a static entity based index seemed to be the most logical rationale for inclusion.  For my personal Use Case, I need time series data because I am trying to build Observability on our ticketing platforms.  This comes with using `ticket.number` as the identifier to link updates to a specific ticket record over time.  (We already have `transaction.id` to link and track an individual transaction update.)  With tickets, the timeframe for change updates over time can be separated by seconds to days.

Anyway, my thinking was that `ticket.created` idenitfied the original ticket creation.  I thought about having `ticket.updated`, but thought that `@timestamp` may also serve that purpose.  @djptek I like your suggestion, I think it is the best of both worlds.